Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/server/VerificationServer.kt

```diff
@@ -86,41 +152,72 @@ class VerificationServer @Inject constructor(
         Timber.tag(TAG).v("retrieveTanFake()")
+        val requiredHeaderPadding = run {
+            var size = HEADER_SIZE_OUR_DATA
+            size -= HEADER_SIZE_OVERHEAD
+
+            // `POST /version/v1/tan`
+            size -= 20
+
+            size
+        }
+        val requiredBodyPadding = run {
+            var size = BODY_SIZE_EXPECTED
+            size -= BODY_SIZE_OVERHEAD
+
+            // `"registrationToken":"63b4d3ff-e0de-4bd4-90c1-17c2bb683a2f",`
+            size -= 59
+
+            size
+        }
+
         val response = api.getTAN(
             fake = "1",
-            headerPadding = requestPadding(PADDING_LENGTH_HEADER_TAN),
+            headerPadding = requestPadding(requiredHeaderPadding),
             requestBody = VerificationApiV1.TanRequestBody(
                 registrationToken = DUMMY_REGISTRATION_TOKEN,
-                requestPadding = requestPadding(PADDING_LENGTH_BODY_TAN_FAKE)
+                requestPadding = requestPadding(requiredBodyPadding)
             )
         )
         Timber.tag(TAG).v("retrieveTanFake() -> %s", response)
         response
     }
 
     companion object {
-        // padding registration token
-        private const val VERIFICATION_BODY_FILL = 139
+        /**
+         * The specific sizes are not important, but all requests should be padded up to the same size.
+         * Pick a total size that is guaranteed to be above or equal to the maximum size a request can be.
+         */
+        // `"requestPadding":""`
+        private const val BODY_SIZE_PADDING_OVERHEAD = 19 //
+
+        // `{}` json brackets
+        private const val BODY_SIZE_OVERHEAD = BODY_SIZE_PADDING_OVERHEAD + 2
+        private const val BODY_SIZE_EXPECTED = 250
+
+        /**
+         * The header itself is larger.
+         * We care about the header fields we set that are request specific.
+         * We don't need to pad for device specific fields set by OK http.
+         */
+        // `POST /version/v1/registrationToken` -> 34 (longest method + url atm) use 64 to have a buffer
+        private const val HEADER_SIZE_LONGEST_METHOD = 34
 
-        const val PADDING_LENGTH_HEADER_REGISTRATION_TOKEN = 0
-        const val PADDING_LENGTH_BODY_REGISTRATION_TOKEN_TELETAN = 51 + VERIFICATION_BODY_FILL
-        const val PADDING_LENGTH_BODY_REGISTRATION_TOKEN_GUID = 0 + VERIFICATION_BODY_FILL
+        // `cwa-fake 0\n` -> 12
+        private const val HEADER_SIZE_VAL_FAKE = 12
 
-        // padding test result
-        const val PADDING_LENGTH_HEADER_TEST_RESULT = 7
-        const val PADDING_LENGTH_BODY_TEST_RESULT = 31 + VERIFICATION_BODY_FILL
+        // `cwa-header-padding\n` -> 22
+        private const val HEADER_SIZE_VAL_PADDING = 22
+        private const val HEADER_SIZE_OVERHEAD = HEADER_SIZE_VAL_FAKE + HEADER_SIZE_VAL_PADDING
+        private const val HEADER_SIZE_OUR_DATA = HEADER_SIZE_LONGEST_METHOD + HEADER_SIZE_OVERHEAD
```

Comment:
We pad the request for plausible deniability, so they all have the same size.
Previously the padding was difficult to understand why a certain value was picked, as we summed up the differences to other API requests, this was a bit too :mage:.
This PR changes the padding calculation to start of with an expected size, and then each API call expects its own size, and what we are left with is the  required padding.